File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/go_solutions/3170-lexicographically-minimum-string-after-removing-stars.go
func removeStars(s string) string {
    stack := make([]rune, 0)

    for _, char := range s {
        if char == '*' {
            if len(stack) > 0 {
                stack = stack[:len(stack)-1]
            }
        } else {
            stack = append(stack, char)
        }
    }

    return string(stack)
}

File: leetcode/AIcoding/2.Gemini-2.0/c_solutions/3170-lexicographically-minimum-string-after-removing-stars.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

char *removeStars(char *s) {
    int n = strlen(s);
    char *stack = (char *)malloc((n + 1) * sizeof(char));
    int top = -1;

    for (int i = 0; i < n; i++) {
        if (s[i] == '*') {
            if (top >= 0) {
                top--;
            }
        } else {
            stack[++top] = s[i];
        }
    }

    stack[top + 1] = '\0';
    char *result = (char *)malloc((top + 2) * sizeof(char));
    strcpy(result, stack);

    free(stack);
    return result;
}